### PR TITLE
Removing some temporary changes needed for yaml testing migration

### PR DESCRIPTION
**Pull request by daineAMD**

### CODE REVIEW

File: clients/common/utility.cpp [hunk 1]
```diff
@@ -218,12 +218,16 @@ hipblasLocalHandle::~hipblasLocalHandle()
...
+        if(hipStatus != hipSuccess)
+            std::cerr << "error freeing hip memory in hipblasLocalHandle: "
+                      << hipGetErrorString(hipStatus) << "\n";
     }
     hipblasStatus_t status = hipblasDestroy(m_handle);
     if(status != HIPBLAS_STATUS_SUCCESS)
     {
-        printf("hipblasDestroy error!\n");
+        std::cerr << "hipblasDestroy error: " << hipblasStatusToString(status) << "\n";
     }
```
Comment (TorreZuk) [inline clients/common/utility.cpp]: After this you might also want to do the hipGetLastError check like I added to rocblas teardown.

File: clients/include/d_vector.hpp [hunk 2]
```diff
@@ -76,13 +77,15 @@ class d_vector
             if(PAD > 0)
             {
                 // Copy guard to device memory before allocated memory
-                CHECK_HIP_ERROR(hipMemcpy(d, guard, sizeof(guard), hipMemcpyHostToDevice));
+                if(hipMemcpy(d, guard, sizeof(guard), hipMemcpyHostToDevice))
+                    std::cerr << "Error: hipMemcpy pre-guard copy failure." << std::endl;
```
Comment (TorreZuk) [inline clients/include/d_vector.hpp:81]: Does this ensure a failed test?

**Review by TorreZuk — APPROVED** [review]
LGTM if total tests match expected count.  Points raised can be addressed later.

File: clients/common/utility.cpp (hunk 1 above)
Comment (daineAMD) [reply clients/common/utility.cpp]: Added hipGetLastError check as suggested, but this is a hipblas failure not hip, so needed to add gtest test

File: clients/include/d_vector.hpp (hunk 2 above)
Comment (daineAMD) [reply clients/include/d_vector.hpp:81]: With hipGetLastError added, now it does.

**Review by TorreZuk — APPROVED** [review]
LGTM

**Pull request merged**